src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_Picomotor8742.py

```diff
@@ -0,0 +1,144 @@
+              {'title': 'IP address:', 'name': 'ip', 'type': 'str','value': "192.168.0.109"},
+              {'title': 'Axis parameters: ', 'name': 'axis_p','type': 'group','children':[
+                  {'title': 'Velocity (steps/s): ', 'name': 'speed_axis','type': 'int','value':0,'min':1,'max':2e3 },
+                  {'title': 'Acceleration (steps/s^2): ', 'name': 'acc_axis', 'type': 'int','value':0,'min':1,'max':2e5},
+                  {'title': 'Type: ', 'name': 'motor', 'type':'str','value':'None'},]}] + comon_parameters_fun(is_multiaxes, axes_names, epsilon=_epsilon)
+    # _epsilon is the initial default value for the epsilon parameter allowing pymodaq to know if the controller reached
+    # the target value. It is the developer responsibility to put here a meaningful value
+
+    def get_actuator_value(self):
+        """Get the current value from the hardware with scaling conversion.
+
+        Returns
+        -------
+        float: The position obtained after scaling conversion.
+        """
+        axis = int(self.settings.child('multiaxes', 'axis').value())
+        pos = self.controller.get_position(axis=axis)
+        return pos
+
+    def close(self):
+        """Terminate the communication """
+        self.controller.close()  # when writing your own plugin replace this line
+
+    def commit_settings(self, param: Parameter):
+        """Apply the consequences of a change of value in the detector settings
+
+        Parameters
+        ----------
+        param: Parameter
+            A given parameter (within detector_settings) whose value has been changed by the user
+        """
+        if param.name() == 'speed_axis' or param.name() == 'acc_axis':
+            axis = int(self.settings.child('multiaxes', 'axis').value())
+            self.controller.setup_velocity(axis=axis, speed=self.settings.child('axis_p','speed_axis').value(),
+                                           accel=self.settings.child('axis_p','acc_axis').value() )
+        else:
+            pass
+
+    def ini_stage(self, controller=None):
+        """Actuator communication initialization
+
+        Parameters
+        ----------
+        controller: (object)
+            custom object of a PyMoDAQ plugin (Slave case). None if only one actuator by controller (Master case)
+
+        Returns
+        -------
+        info: str
+        initialized: bool
+            False if initialization failed otherwise True
+        """
+
+        self.controller = self.ini_stage_init(slave_controller=controller)
+
+        if self.is_master:
+            self.controller = Newport.Picomotor8742(self.settings.child('ip').value())
+
+        try:
+            info = self.controller.get_id()
+            initialized = True
+        except:
+            info = ""
+            initialized = False
+
+        if initialized:
+            motor_types = self.controller.autodetect_motors()
+            axis = int(self.settings.child('multiaxes', 'axis').value())
```

Comment:
here you're looking at the new property:

```
self.axis_index_key
```

However, I would suggest you change the def of your axes to: 

```
axes_names = dict(1=1, 2=2, 3=3, 4=4]
```

then whenever you call self.axis_name you'll get the current key of the dict (string) and if you call self.axis_value,you'll get the current value in the dict (the int)